src/plugins/orangehrmAuthenticationPlugin/Controller/LoginController.php

```diff
@@ -55,6 +61,17 @@ public function getHomePageService(): HomePageService
         return $this->homePageService;
     }
 
+    /**
+     * @return SocialMediaAuthenticationService
+     */
+    public function getSocialMediaAuthenticationService(): SocialMediaAuthenticationService
+    {
+        if (!$this->socialMediaAuthenticationService instanceof SocialMediaAuthenticationService) {
+            $this->socialMediaAuthenticationService = new SocialMediaAuthenticationService();
+        }
+        return $this->socialMediaAuthenticationService;
```

Comment:
```suggestion
        return $this->socialMediaAuthenticationService ??= new SocialMediaAuthenticationService();
```
:thinking: 
OR
why we are not getting from the service container